yt_dlp/extractor/archiveorg.py

```diff
@@ -285,38 +285,99 @@ class YoutubeWebArchiveIE(InfoExtractor):
             }
         },
         {
-            # Old flash-only video
+            # Old flash-only video. Webpage title starts with "YouTube - ".
             'url': 'https://web.archive.org/web/20081211103536/http://www.youtube.com/watch?v=jNQXAC9IVRw',
             'info_dict': {
                 'id': 'jNQXAC9IVRw',
                 'ext': 'unknown_video',
-                'title': 'jNQXAC9IVRw'
+                'title': 'Me at the zoo'
             }
+        },
+        {   # Some versions of Youtube have have "Youtube" as page title in html (and later rewritten by js).
+            'url': 'https://web.archive.org/web/http://www.youtube.com/watch?v=kH-G_aIBlFw',
+            'info_dict': {
+                'id': 'kH-G_aIBlFw',
+                'ext': 'mp4',
+                'title': 'kH-G_aIBlFw'
+            },
+            'expected_warnings': [
+                'Unable to extract video title',
+            ]
+        },
+        {
+            # First capture is a 302 redirect intermediary page.
+            'url': 'https://web.archive.org/web/20050214000000/http://www.youtube.com/watch?v=0altSZ96U4M',
+            'info_dict': {
+                'id': '0altSZ96U4M',
+                'ext': 'mp4',
+                'title': '0altSZ96U4M'
+            },
+            'expected_warnings': [
+                'Unable to extract video title',
+            ]
+        },
+        {
+            # Video not archived, only capture is unavailable video page
+            'url': 'https://web.archive.org/web/20210530071008/https://www.youtube.com/watch?v=lHJTf93HL1s&spfreload=10',
+            'only_matching': True,
         }
     ]
 
     def _real_extract(self, url):
         video_id = self._match_id(url)
+        title = video_id  # if we are not able get a title
+
+        def _extract_title(webpage):
+            page_title = self._html_search_regex(r'<title>([^<]*)</title>', webpage, 'title', fatal=False) or ''
+            page_title = page_title.strip()
+            # Unavailable video captures will only have youtube suffix/prefix as page title.
+            if page_title.endswith("- YouTube"):
+                page_title = page_title[:-9].strip()
+            elif page_title.startswith("YouTube -"):
+                page_title = page_title[9:].strip()
+            else:
+                # Probably invalid title if doesn't have youtube suffix/prefix
+                self.report_warning("Unable to extract video title", video_id=video_id)
+                return
+            if not page_title:
+                self.report_warning("Unable to extract video title", video_id=video_id)
+                return
+            return page_title
+
+        # If the video is no longer available,
+        # the oldest capture may be one before it was removed
+        webpage = self._download_webpage(
+            "https://web.archive.org/web/20050214000000/http://www.youtube.com/watch?v=%s" % video_id,
```

Comment:
From my observations, setting this to an early date will make it redirect to the closest date from there.